src/query/storages/fuse/src/operations/util.rs

```diff
@@ -73,3 +82,39 @@ pub fn column_parquet_metas(
     }
     Ok(col_metas)
 }
+
+/// Checks if a duplicate label exists in the meta store.
+///
+/// # Arguments
+///
+/// * `ctx` - The table context. Must implement the `TableContext` trait and be wrapped in an `Arc`.
+///
+/// # Returns
+///
+/// Returns a `Result` containing a `bool` indicating whether specific duplicate label exists (`true`) or not (`false`).
+pub async fn check_duplicate_label(ctx: Arc<dyn TableContext>) -> Result<bool> {
+    let duplicate_label = ctx.get_settings().get_duplicate_label().unwrap();
+    if duplicate_label.is_empty() {
+        Ok(false)
+    } else {
+        let kv_store = UserApiProvider::instance().get_meta_store_client();
+        let raw = kv_store.get_kv(&duplicate_label).await?;
+        match raw {
+            None => {
+                let expire_at = SeqV::<()>::now_ms() / 1000 + 24 * 60 * 60 * 1000;
+                let _ = kv_store
+                    .upsert_kv(UpsertKV {
```

Comment:
check should be checked before the whole query pipeline.

upsert should be committed with the update meta commit.